lib/jsdom/living/nodes/Document-impl.js

```diff
@@ -196,6 +197,12 @@ class DocumentImpl extends NodeImpl {
     // https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#throw-on-dynamic-markup-insertion-counter
     this._throwOnDynamicMarkupInsertionCounter = 0;
 
+    // Create DOMSelector instance
+    this._domSelector = new DOMSelector(this._globalObject, this._ownerDocument, {
+      domSymbolTree,
+      idlUtils
+    });
```

Comment:
This moves us from lazily initializing nwsapi/domSelector to always initializing them.

Is that OK? The `DOMSelector` constructor looks cheap but I wonder if the nwsapi initialization is expensive.

It's certainly much simpler to just always initialize, so I like your new code for simplicity. But it might be worth checking the impact on very simple benchmarks like `benchmark/dom/construction.js`.

Reply:
Should be OK.

* I modified @asamuzakjp/nwsapi not to assume document.documentElement at install time.
  Instead look for document.documentElement at runtime when needed.
  https://www.npmjs.com/package/@asamuzakjp/nwsapi?activeTab=code L7
* On dom-selector side, it does not pass selector to @asamuzakjp/nwsapi if document.documentElement is not present.
  https://github.com/asamuzaK/domSelector/blob/main/src/index.js#L73

Reply:
Right, I guess the lazy initialization was not for performance, but due to the nwsapi architecture. Thanks!